pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -343,8 +350,99 @@ func (srv AlertmanagerSrv) RoutePostTestReceivers(c *contextmodel.ReqContext, bo
 }
 
 func (srv AlertmanagerSrv) RoutePostTestTemplates(c *contextmodel.ReqContext, body apimodels.TestTemplatesConfigBodyParams) response.Response {
-	// TODO: Stub
-	return response.JSON(http.StatusOK, apimodels.TestTemplatesResults{})
+	am, errResp := srv.AlertmanagerFor(c.OrgID)
+	if errResp != nil {
+		return errResp
+	}
+
+	paths := make([]string, 0)
+	for _, name := range am.GetStatus().Config.Templates {
```

Comment:
I don't think this works as expected? The following template called **other** doesn't return test2 or test3.

```
{{ define "test2" }}
test2
{{ end }}

{{ define "test3" }}
test3
{{ end }}
```